operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -423,10 +447,150 @@ func createManifestWorks(
+	// Delete hub endpoint operator
+	err := c.Delete(context.TODO(), &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      config.HubEndpointOperatorName,
+			Namespace: config.GetDefaultNamespace(),
+		},
+	})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		log.Error(err, "Failed to delete hub endpoint operator")
+		return err
+	}
+	for _, name := range []string{config.HubUwlMetricsCollectorName, config.HubMetricsCollectorName} {
+		err := c.Delete(context.TODO(), &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: config.GetDefaultNamespace(),
+			},
+		})
+		if err != nil && !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to delete hub metrics-collector deployment")
+			return err
+
+		}
+	}
+	hubMetricCollectorSecrets := []string{operatorconfig.HubMetricsCollectorMtlsCert, managedClusterObsCertName, operatorconfig.HubInfoSecretName, config.AlertmanagerAccessorSecretName}
+	for _, name := range hubMetricCollectorSecrets {
+		err := c.Delete(context.TODO(), &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: config.GetDefaultNamespace(),
+			},
+		})
+		if err != nil && !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to delete hub metrics-collector secret")
+			return err
+		}
+	}
+	hubMetricsCollectorConfigMaps := []string{operatorconfig.ImageConfigMap, operatorconfig.CaConfigmapName}
+	for _, name := range hubMetricsCollectorConfigMaps {
+		err := c.Delete(context.TODO(), &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: config.GetDefaultNamespace(),
+			},
+		})
+		if err != nil && !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to delete hub metrics-collector configmap")
+			return err
+		}
+
+	}
+	err = DeleteHubMonitoringClusterRoleBinding(context.TODO(), c)
+	if err != nil {
+		log.Error(err, "Failed to delete monitoring cluster role binding for hub metrics collection")
+		return err
+	}
+	err = DeleteHubCAConfigmap(context.TODO(), c)
+	if err != nil {
+		log.Error(err, "Failed to delete CA configmap for hub metrics collection")
+		return err
+	}
+	err = RevertHubClusterMonitoringConfig(context.TODO(), c)
+	if err != nil {
+		log.Error(err, "Failed to revert cluster monitoring config")
+		return err
+	}
+
+	//isHypershift := true
+	//if os.Getenv("UNIT_TEST") != "true" {
+	//	crdClient, err := util.GetOrCreateCRDClient()
+	//	if err != nil {
+	//		log.Error(err, "Failed to create CRD client")
+	//		return err
+	//	}
+	//	isHypershift, err = util.CheckCRDExist(crdClient, "hostedclusters.hypershift.openshift.io")
+	//	if err != nil {
+	//		log.Error(err, "Failed to check if the CRD hostedclusters.hypershift.openshift.io exists")
+	//		return err
+	//	}
+	//}
+	//if isHypershift {
+	//	err = DeleteServiceMonitors(context.TODO(), c)
+	//	if err != nil {
+	//		log.Error(err, "Failed to delete service monitors for hub metrics collection")
+	//		return err
+	//	}
+	//}
```

Comment:
Delete this?

Reply:
Sorry this was me, trying to get it to work